### DISCUSSION TITLE
Defining foreign keys on composite primary keys

### DISCUSSION THREAD

**Discussion by spauldhaliwal** [simolus3/drift#2345] [Q&A] [answered]
When using drift , we can define a foreign key like so `IntColumn get author => integer().references(Authors, #id)();`, but how do we make references to composite primary keys?

_ex defining a foreign key referenece on:_

```
@override
Set<Column> get primaryKey => {id, clientIdentifier};
```

**Comment by simolus3** [comment, marked as answer]
For Dart defined tables, the only way to define a composite foreign key is to override [customConstraints](https://drift.simonbinder.eu/api/drift/table/customconstraints) on `Table`:

```dart
class References extends Table {
  IntColumn get id => integer()();
  TextColumn group => text()();

  @override
  get primaryKey => {id, group};
}

class MyTable extends Table {
  IntColumn get id => integer()();
  TextColumn group => text()();

  @override
  List<String> get customConstraints => [
    'FOREIGN KEY (id, group) REFERENCES "references" (id, group)',
  ];
}
```
